NOnion/Services/TorServiceHost.fs

```diff
@@ -112,10 +112,7 @@ type TorServiceHost
                             introEncPrivKey
                             introEncPubKey
                 with
-                | :? TimeoutException
-                | :? CircuitDestroyedException
-                | :? CircuitTruncatedException
-                | :? GuardConnectionFailedException
+                | :? NOnionException
```

Comment:
last thing to do here: you can extract the retry logic into a Retry method that receives: delegates it needs, retry count, retry max, and two types of exceptions as generic types; this way this method will be much more readable